src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -163,50 +171,38 @@ public void SetDefaultHubName(string hubName)
             this.defaultHubName = hubName;
         }
 
-        internal string GetDebugString()
+        internal void TraceConfiguration(EndToEndTraceHelper traceHelper)
         {
-            var sb = new StringBuilder(4096);
-            sb.AppendLine("Initializing extension with the following settings:");
-            sb.Append(nameof(this.HubName)).Append(":").Append(this.HubName).Append(", ");
-
-            this.AppendStorageProviderValuesToDebugString(sb);
-
-            sb.Append(nameof(this.MaxConcurrentActivityFunctions)).Append(": ").Append(this.MaxConcurrentActivityFunctions).Append(", ");
-            sb.Append(nameof(this.MaxConcurrentOrchestratorFunctions)).Append(": ").Append(this.MaxConcurrentOrchestratorFunctions).Append(", ");
-            sb.Append(nameof(this.ExtendedSessionsEnabled)).Append(": ").Append(this.ExtendedSessionsEnabled).Append(", ");
-            sb.Append(nameof(this.UseGracefulShutdown)).Append(": ").Append(this.UseGracefulShutdown).Append(", ");
-            if (this.ExtendedSessionsEnabled)
-            {
-                sb.Append(nameof(this.ExtendedSessionIdleTimeoutInSeconds)).Append(": ").Append(this.ExtendedSessionIdleTimeoutInSeconds).Append(", ");
-            }
-
-            if (this.NotificationUrl != null)
+            // Clone the options to avoid making changes to the original.
+            // We make updates to the clone rather than to JSON to ensure we're updating what we think we're updating.
+            DurableTaskOptions clone = JObject.FromObject(this).ToObject<DurableTaskOptions>();
```

Comment:
@ConnorMcMahon just a heads-up on this change. Through manual testing I discovered that I wasn't correctly overwriting existing config values. I was instead adding new values (different capitalization) to the trace output. To avoid issues like this, I decided to switch from modifying JSON data to modifying the Options object directly (or rather, a clone of the options object) and then serializing the updated version. That way this tracing code never needs to be concerned with the details of JSON serialization.